src/zenml/zen_stores/base_zen_store.py

```diff
@@ -631,6 +633,88 @@ def get_role_assignments_for_team(
             KeyError: If no team or project with the given names exists.
         """
 
+    @property
+    @abstractmethod
+    def store_associations(self) -> List[StoreAssociation]:
+        """Fetches all artifact/metadata store associations.
+
+        Returns:
+            A list of all artifact/metadata store associations.
+        """
+
+    @abstractmethod
+    def create_store_association(
```

Comment:
do all these new methods need to be abstract? I feel like most subclasses will perform the same sort of logic here, i.e., `rest_zen_store` and `local_zen_store` seem to have a lot of code duplication in these methods. Maybe there could be a standard implementation that only stores with weird custom syntax (e.g. SQL) need to overwrite?